### DISCUSSION TITLE
[Question][Hw7] Scope of hw7

### DISCUSSION THREAD

**Discussion by jirheee** [prosyslab-classroom/cs524-program-analysis#337] [Q&A] [answered]
According to [Llvm documentation about ICmp instruction](https://llvm.org/docs/LangRef.html#:~:text=The%20remaining%20two%20arguments%20must%20be%20integer%20or%20pointer%20or%20integer%20vector%20typed.%20They%20must%20also%20be%20identical%20types.), pointers can also be an operand. Should we allow pointer comparisons?

> The remaining two arguments must be [integer](https://llvm.org/docs/LangRef.html#t-integer) or [pointer](https://llvm.org/docs/LangRef.html#t-pointer) or integer [vector](https://llvm.org/docs/LangRef.html#t-vector) typed. They must also be identical types.

**Comment by jirheee** [comment]
Also, should we take care of `ptrtoint` cast of llvm?

**Comment by goodtaeeun** [comment, marked as answer]
For the first question, pointer is a type you should support, and `icmp` is an instruction you should support. So yes, you should allow pointer comparisons.
For the second questions, any of the llvm instructions not mentioned in https://github.com/prosyslab-classroom/smallvm-typechecker-master#3-format-of-input-programs is considered `nop`. 

Hope this clarifies your question.

**Comment by jirheee** [comment]
I think the page is not accessible...?

**Comment by goodtaeeun** [comment]
Oh, sorry. It is the third section of the Readme of the HW7
